### PR TITLE
Fix 100 :: Add listener to delete EasyTag and TagCheck records on channel deletion

### PR DESCRIPTION
**Pull request by QuantumGhost7**
Added an `on_channel_delete` event listener that checks for EasyTag and TagCheck records associated with the deleted channel and deletes them if found.

### CODE REVIEW

**Comment by deadaf** [comment]
Though this seems right but I would like to mention a a change you should make,
- Tagcheck and Eztag can only be set for a TextChannel. With the correct type hint, we can prevent extra calls to the database, for ex. when a voice channel is deleted.

**Review by deadaf — CHANGES_REQUESTED** [review]
Though this seems right but I would like to mention a a change you should make,

Tagcheck and Eztag can only be set for a TextChannel. With the correct type hint, we can prevent extra calls to the database, for ex. when a voice channel is deleted.

**Comment by QuantumGhost7** [comment]
Thanks for pointing this out, added a commit to make the listener handle the deletion of TextChannel objects only

File: src/cogs/esports/events/tags.py [hunk 1]
```diff
@@ -111,3 +111,18 @@ async def on_eztag_msg(self, message: discord.Message):
             if eztag.delete_after:
                 self.bot.loop.create_task(delete_denied_message(message, 60))
                 self.bot.loop.create_task(delete_denied_message(msg, 60))
+
+    @Cog.listener(name="on_guild_channel_delete")
+    async def on_channel_delete(self, channel: discord.TextChannel):
```
Comment (rtk-rnjn) [inline src/cogs/esports/events/tags.py:116]: I don't think, dispatch works in that way, instead why not checking channel type inside the function
```py
async def on_channel_delete(self, channel: discord.abc.GuildChannel) -> None:
    if not isinstance(channel, discord.TextChannel):
        return
    ...
```

Comment (QuantumGhost7) [reply src/cogs/esports/events/tags.py]: Updated the PR

**Comment by rtk-rnjn** [comment]
This change is really good in terms of maintaining clean database, but catch is Q sees about 5Lac+ text channels, on every deletion of channel database will be called.

🤔 A caching mechanism or some periodic asyncio task would be better. Thats dirty, let Rohit decide...

**Comment by deadaf** [comment]
@rtk-rnjn Now that you mention it, I recalled we do have it already. Apologies, It was a fault on my side @QuantumGhost7 , I forgot to mention it earlier. 
We have cache system in https://github.com/quotientbot/Quotient-Bot/blob/main/src/core/cache.py
in the listener `self.bot.cache.tagcheck` is the set of channel ids for tagcheck channels & `self.bot.cache.eztagchannels` for eztag. 

We should just check if channel id is in the cache set , if it is directly make the delete call, no need to check first since it avoids the additional round trip, then just remove the channel id from cache.

pls let me know if you are free to make the changes @QuantumGhost7 .

**Comment by QuantumGhost7** [comment]
Added a commit. Also would it be better if I create a new PR, on another branch of the forked repo and close this one, just to avoid merging the previous commits.

**Review by deadaf — COMMENTED** [review]
Yes, I guess a new PR should be fine, also since we are checking the channel id from the cache only, no need to check if the record is present or not, directly make the delete call. This will avoid a round trip to the db.

**Pull request closed without merging**